### ISSUE TITLE
passing multiple losses and combining them

### ISSUE THREAD

**Issue by mithunpaul08** [UKPLab/sentence-transformers#234]
Hi,

I had a qn. Not sure if it had to be posted as an issue or discussion. Sorry.

The qn I had was: Can I use train_objectives iterator (given to model.fit) to pass a bunch of different [data_loader, loss] pairs?  Can you point me to an example in your repo where you do that?Sorry, I couldn't find much of a documentation for train_objectives. Also do you use combined losses anywhere? So far I have only gone through training_nli.py and its associated classes like SentenceTransformer etc- hence kindly pardon my ignorance

model.fit(train_objectives=[(train_dataloader, train_loss)],

Long version/background story:
Am trying to recreate a teacher student [model](http://papers.nips.cc/paper/6719-mean-teachers-are-better-role-models-weight-averaged-consistency-targets-improve-semi-supervised-deep-learning-results.pdf) for nli with bert/sentence-transformers. I am trying to train a student model using a teacher model

The idea is that: the student and teacher models will be bert models internally, but will be reading two different nli datasets. And they will both be connected by a combined loss function.

**Comment by nreimers** [comment]
Have a look here:
https://github.com/UKPLab/sentence-transformers/blob/master/examples/training_transformers/training_multi-task.py

This might also be helpful:
https://github.com/UKPLab/sentence-transformers/blob/master/docs/pretrained-models/multilingual-models.md

There, I use a teacher model to derive a multilingual model. This setup could of course also be used for other things

**Comment by mithunpaul08** [comment]
aaah beautiful...thanks a lot @nreimers
